src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,29 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private final int userPasswordLength = 6;
+    private final int userAge = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            try {
+                throw new UserAlreadyExistsException(user.getLogin());
+            } catch (UserAlreadyExistsException e) {
+                System.out.println(e.toString());
+            }
+        }
+        if (user.getPassword().length() < userPasswordLength) {
+            try {
+                throw new PasswordLengthException(6);
+            } catch (PasswordLengthException e) {
+                System.out.println(e.toString());
+            }
+        }
+        if (user.getAge() < userAge) {
+            throw new IllegalArgumentException("The Age of user is not valid");
+        }
+        return user;
```

Comment:
Check parameters for null at first, and after this to their value or length, don`t wrap throwing your exceptions